File: benchmark/languages/go/work/day01/test_edge.go
package main

import "fmt"

func countZerosDuring(pos, dist int, dir string) int {
	var firstI int
	if dir == "L" {
		if pos == 0 {
			firstI = 100
		} else {
			firstI = pos
		}
	} else {
		firstI = (100 - pos) % 100
		if firstI == 0 {
			firstI = 100
		}
	}
	if firstI > dist {
		return 0
	}
	return (dist-firstI)/100 + 1
}

func main() {
	// R1000 from 50 should hit 0 ten times
	fmt.Println("R1000 from 50:", countZerosDuring(50, 1000, "R"))
	// final pos = (50 + 1000) % 100 = 1050 % 100 = 50, so it returns to 50 hitting 0 ten times
	
	// Verify: first hit at i=50, then 150, 250, 350, 450, 550, 650, 750, 850, 950
	// That's 10 hits. (1000-50)/100 + 1 = 950/100 + 1 = 9 + 1 = 10 ✓
}

File: benchmark/languages/c/work/day01/test_edge.c
#include <stdio.h>
// Test: R1000 from 50 should pass through 0 exactly 10 times
// first_k = 100 - 50 = 50
// count = (1000 - 50) / 100 + 1 = 950/100 + 1 = 9 + 1 = 10
int main() {
    int pos = 50, dist = 1000;
    int first_k = (pos == 0) ? 100 : (100 - pos);
    long long count = 0;
    if (dist >= first_k) count = (long long)(dist - first_k) / 100 + 1;
    printf("R1000 from 50: passes through 0 = %lld times (expected 10)\n", count);
    // New position: (50 + 1000) % 100 = 1050 % 100 = 50
    printf("New position: %d (expected 50)\n", (50 + 1000) % 100);
    return 0;
}
